packages/runtime/src/decorators/response.ts

```diff
@@ -1,10 +1,22 @@
-export function SuccessResponse(name: string | number, description?: string): Function {
+export function SuccessResponse(name: string | number, description?: string): Function;
+
+export function SuccessResponse<T>(name: string | number, description?: string): Function;
+
+export function SuccessResponse<T>(name: string | number, description?: string): Function {
   return () => {
     return;
   };
 }
 
-export function Response<T>(name: string | number, description?: string, example?: T): Function {
+export function Response(name: string | number, description?: string): Function;
+
+export function Response<U>(name: string | number, description?: string): Function;
+
+export function Response<T>(name: string | number, description?: string, example?: T): Function;
+
+export function Response<T, U>(name: string | number, description?: string, example?: T): Function;
+
+export function Response<T, U>(name: string | number, description?: string, example?: T): Function {
```

Comment:
I think the goal here is to provide options, but I'm afraid these are too many.
This will cover what we need in a way that's not ambiguous:

```ts
export function SuccessResponse<HeaderType>(name: string | number, description?: string): Function {
  return () => {
    return;
  };
}

export function Response<ExampleType, Header = {}>(name: string | number, description?: string, example?: ExampleType): Function {
  return () => {
    return;
  };
}
```

Otherwise, it looks like this:

```ts
export function Response<U>(name: string | number, description?: string): Function;

export function Response<T>(name: string | number, description?: string, example?: T): Function;
```

means the argument can be only the Header type, but the parsing will just fo the wrong thing.